integration-tests/tests/src/node/custom-fetch.ts

```diff
@@ -0,0 +1,66 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { expect } from "chai";
+import { importAppBefore } from "../hooks";
+import { buildAppConfig } from "../utils/build-app-config";
+import fetch, { Request } from "node-fetch";
+import Realm from "realm";
+
+describe("custom fetch", () => {
+  let called = false;
+
+  const HTTP_METHOD: Record<number, string> = {
+    [0]: "GET",
+    [1]: "POST",
+    [2]: "PUT",
+    [3]: "PATCH",
+    [4]: "DELETE",
+  };
+
+  const customFetch = async (request: Request) => {
+    called = true;
+    const { url, body, ...rest } = request;
+    if (typeof request.method == "number") {
+      rest.method = HTTP_METHOD[request.method];
+    }
+    const response = await fetch(url, rest);
+    return response;
+  };
```

Comment:
An alternative could be to implement this as a wrapping of the default `fetch` from `@realm/network-transport`.

Reply:
As the user is requesting overriding the `AbortController`, it's probably worth while being full flexibly

Reply:
I think this solution is good, otherwise we would also have to provide a mechanism to override the default `AbortController` as well.  This basically lets the user override `@realm/network-transport`.
I guess the crux of the issue, is the user that reported the original issue is using `electron` and this is using the node bindings instead of the browser bindings to `fetch`.